sycl/doc/extensions/proposed/sycl_ext_intel_cache_controls.asciidoc

```diff
@@ -92,178 +92,217 @@ supports.
 
-template<cache_level L>
-inline constexpr cache_control_read_key::value_t<cache_control_read_type::cached, L>
-cache_control_read_cached;
+Repeating a cache level within a cache control is an error. For example:
+```c++
+read_hint<cache_control<cache_mode::cached, cache_level::L1, cache_level::L1>>
+```
 
-template<cache_level L>
-inline constexpr cache_control_read_key::value_t<cache_control_read_type::uncached, L> cache_control_read_uncached;
+Specifying more than one cache mode from `read_hint`/`read_assertion`
+or more than one `write_hint` type at a particular cache level is an error.
+For example, specifying `cached` and `uncached` at level L2:
+```c++
+read_hint<cache_control<cache_mode::uncached, cache_level::L2>,
+          cache_control<cache_mode::cached, cache_level::L2>>
+```
+However, a cache mode from `read_hint`/`read_assertion` and
+one from  `write_hint` may be specified at the same level:
+```c++
+read_hint<cache_control<cache_mode::uncached, cache_level::L1>>,
+write_hint<cache_control<cache_mode::streaming, cache_level::L1>>
+```
 
-template<cache_level L>
-inline constexpr cache_control_read_key::value_t<cache_control_read_type::streaming, L> cache_control_read_streaming;
+==== Read hint property
+This property is a hint requesting specific cache behavior when
+loading from memory through an `annotated_ptr`. This property
+can affect the performance of device code, but it does not change
+the semantics.
 
-template<cache_level L>
-inline constexpr cache_control_read_key::value_t<cache_control_read_type::invalidate_after_read, L> cache_control_invalidate_after_read;
+```c++
+namespace sycl::ext {
+namespace intel::experimental {
 
-template<cache_level L>
-inline constexpr cache_control_read_key::value_t<cache_control_read_type::const_cached, L> cache_control_read_const_cached;
+struct read_hint_key {
+  template <typename... Cs>
+  using value_t = property_value<read_hint_key, Cs...>;
+};
 
-template<cache_level L>
-inline constexpr cache_control_write_key::value_t<cache_control_write_type::uncached, L> cache_control_write_uncached;
+template <typename... Cs>
+inline constexpr read_hint_key::value_t<Cs...> read_hint;
 
-template<cache_level L>
-inline constexpr cache_control_write_key::value_t<cache_control_write_type::write_streaming, L> cache_control_write_streaming;
+} // namespace intel::experimental
 
-template<cache_level L>
-inline constexpr cache_control_write_key::value_t<cache_control_write_type::write_through, L> cache_control_write_through;
+namespace oneapi {
+namespace experimental {
 
-template<cache_level L>
-inline constexpr cache_control_write_key::value_t<cache_control_write_type::write_back, L> cache_control_write_back;
+using read_hint_key = intel::experimental::read_hint_key;
+template <> struct is_property_key<read_hint_key> : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<read_hint_key, annotated_arg<T, PropertyListT>>
+    : std::true_type {};
```

Comment:
I don't think we should specialize the `annotated_arg` property.  The `read_hint` property is not allowed in `annotated_arg`.  Same goes for `write_hint` and `read_assertion`.

Reply:
Removed annotated_arg.